src/tests/ftest/util/command_utils.py

```diff
@@ -183,32 +215,63 @@ def stop(self):
 
         """
         if self._process is not None:
-            # Use a list to send signals to the process with one second delays:
-            #   Interupt + wait 3 seconds
-            #   Terminate + wait 2 seconds
-            #   Quit + wait 1 second
-            #   Kill
-            signal_list = [
-                signal.SIGINT, None, None, None,
-                signal.SIGTERM, None, None,
-                signal.SIGQUIT, None,
-                signal.SIGKILL]
+            # Send a SIGTERM to the stop the subprocess and if it is still
+            # running after 5 seconds send a SIGKILL and report an error
+            signal_list = [signal.SIGTERM, signal.SIGKILL]
 
             # Turn off verbosity to keep the logs clean as the server stops
             self._process.verbose = False
 
-            # Keep sending signals and or waiting while the process is alive
+            # Send signals while the process is still running
+            state = None
             while self._process.poll() is None and signal_list:
-                signal_type = signal_list.pop(0)
-                if signal_type is not None:
-                    self._process.send_signal(signal_type)
+                signal_to_send = signal_list.pop(0)
+                msg = "before sending signal {}".format(signal_to_send)
+                state = self.get_subprocess_state(msg)
+                self.log.info(
+                    "Sending signal %s to %s (state=%s)", str(signal_to_send),
+                    self._command, str(state))
+                self._process.send_signal(signal_to_send)
                 if signal_list:
-                    time.sleep(1)
+                    time.sleep(5)
+
             if not signal_list:
-                # Indicate an error if the process required a SIGKILL
-                raise CommandFailure("Error stopping '{}'".format(self))
+                if state and len(state) == 1 and state[0] not in ("D", "Z"):
+                    # Indicate an error if the process required a SIGKILL and
+                    # either multiple processes were still found running or the
+                    # parent process was in any state accept uninterruptible
+                    # sleep (D) or zombie (Z).
+                    raise CommandFailure("Error stopping '{}'".format(self))
+
+            self.log.info("%s stopped successfully", self.command)
             self._process = None
 
+    def get_subprocess_state(self, message=None):
+        """Display the state of the subprocess.
+
+        Args:
+            message (str, optional): additional text to include in output.
+                Defaults to None.
+
+        Returns:
```

Comment:
This method seems to only get the process state associated with the parent process running locally. So should the "(first entry should be the parent process)" part of the comment be removed?

Also, are there any conditions in which state will exceed list length of 1?

Reply:
Yes it does only return the states of the locally running process, but due to the `ps --forest` there may be multiple states that are captured from the output.  So far in testing I've only seen one process match for `ps --forest orteun`.